File: server/src/main/java/com/example/schoolsite/services/impl/PupilServiceImpl.java
package com.example.schoolsite.services.impl;

import com.example.schoolsite.dto.PupilDTO;
import com.example.schoolsite.entity.Classroom;
import com.example.schoolsite.entity.Parents;
import com.example.schoolsite.entity.Pupil;
import com.example.schoolsite.map.Mapper;
import com.example.schoolsite.services.PupilService;
import com.example.schoolsite.workWithDatabase.repo.ClassroomRepository;
import com.example.schoolsite.workWithDatabase.repo.ParentsRepository;
import com.example.schoolsite.workWithDatabase.repo.PupilRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class PupilServiceImpl implements PupilService {

    private final PupilRepository pupilRepository;

    private final ParentsRepository parentsRepository;

    private final ClassroomRepository classroomRepository;

    @Override
    public PupilDTO getPupilByFIO(String userId) {
        Pupil pupil = pupilRepository.findByUserId(Long.parseLong(userId));
        Parents parents = parentsRepository.getById(pupil.getParentsId().getId());
        Classroom classroom = classroomRepository.getById(pupil.getClassroomId().getId());
        return Mapper.mapToPupilDTO(pupil, parents, classroom);
    }
}

File: server/src/main/java/com/example/schoolsite/services/impl/PupilServiceImpl.kt
package com.example.schoolsite.services.impl

import com.example.schoolsite.dto.PupilDTO

@Service
@RequiredArgsConstructor
class PupilServiceImpl : PupilService {
    private val pupilRepository: PupilRepository? = null
    private val parentsRepository: ParentsRepository? = null
    private val classroomRepository: ClassroomRepository? = null
    @Override
    fun getPupilByFIO(userId: String?): PupilDTO {
        val pupil: Pupil = pupilRepository.findByUserId(Long.parseLong(userId))
        val parents: Parents = parentsRepository.getById(pupil.getParentsId().getId())
        val classroom: Classroom = classroomRepository.getById(pupil.getClassroomId().getId())
        return Mapper.mapToPupilDTO(pupil, parents, classroom)
    }
}